SingleSource/UnitTests/AArch64/CMakeLists.txt

```diff
@@ -1,10 +1,9 @@
 if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
+  if(TARGET_OS STREQUAL "Linux")
+    list(APPEND CFLAGS --rtlib=compiler-rt)
+    list(APPEND LDFLAGS --rtlib=compiler-rt)
```

Comment:
Guessing this might be an issue if the compiler does include / has not built compiler-rt.

Reply:
Thank you for report, compiler-rt is required for these tests to work on Linux. Is there a clang configuration without compiler-rt used? Not sure do we need to detect in the tests that compiler-rt is missing and not include them in testing or just make compiler-rt build part of buildbot configuration.

Reply:
I found the rest of the linker output:
```
/usr/bin/ld: cannot find /home/tcwg-buildbot/worker/clang-aarch64-global-isel/stage1.install/lib/clang/19/lib/linux/libclang_rt.builtins-aarch64.a: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Which confirms it.

There's no requirement to have compiler-rt present and it seems like this is the first set of tests that need it. A script was recently added to find it on Windows https://github.com/llvm/llvm-test-suite/commit/1c4054cda38233a71767832048a28a9e4c41adaa so I take from that that it has never been needed on Linux before.

So if that script also works for Linux you could call that and skip if we don't find it. @omjavaid would that work on Linux too?

The other option is a "does compiler support --rtlib=compiler-rt" check, in the normal cmake fashion.

(and coverage wise, we have other AArch64 bots that do build compiler-rt that are running the test, so not running it on this one isn't a problem)

Reply:
Yes I think find_compiler_rt_library in llvm-test-suite/cmake/modules/HandleCompilerRT.cmake should work for linux. The function is derived from llvm-project/cmake/modules/HandleCompilerRT.cmake which works for Linux as well. some minor changes are may be required and can be referenced from to the llvm-project version of HandleCompilerRT.cmake module